spring-cloud-gcp-starters/spring-cloud-gcp-starter-core/src/main/java/org/springframework/cloud/gcp/core/autoconfig/GcpPropertiesAutoConfiguration.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2017 original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.core.autoconfig;
+
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * @author João André Martins
+ */
+@Configuration
+@EnableConfigurationProperties(GcpProperties.class)
+public class GcpPropertiesAutoConfiguration {
```

Comment:
Nope....

I think it will be more productive accepting this, and fixing it to show what we meant by properties and configuration.

I think the auto configuration concept is not clear to you it. You don't need one auto configuration file for every properties file, or one autoconfiguration file for every feature.

I guess the best course is to rename the CredentialsAutoConfiguration to be something like GoogleCloudContextAutoConfiguration.

I'll take this on another PR, and clean a bit the project